lib/user/user-client.js

```diff
@@ -29,13 +31,67 @@ var xhr = function(_options) {
   return deferred.promise;
 };
 
-var storeProfile = function(profileData) {
-  localStorage.setItem('fh.wfm.profileData', JSON.stringify(profileData));
+/**
+* Encrypt the users profile data using AES with a hash of the session key and store the result in localStorage.
+* @param profileData - the plaintext to encrypt.
+* @param sessionToken - the secret key to encrypt with.
+*/
+var storeProfile = function(profileData, sessionToken) {
+  var deferred = q.defer();
+  if (sessionToken !== null && profileData !== null) {
+    sha256.hash(sessionToken).then(function(hash) {
+      aes.encrypt(profileData, hash).then(function(ciphertext) {
+        localStorage.setItem('fh.wfm.profileData', ciphertext);
+        return deferred.resolve();
+      }, function(err) {
+        console.log("Could not encrypt the data.", err);
+        localStorage.setItem('fh.wfm.profileData', null);
+        return deferred.reject();
+      }
+      );
+    }, function(err) {
+      console.log("Could not generate hash value for the session token", err);
+      localStorage.setItem('fh.wfm.profileData', null);
+      return deferred.reject();
+    });
+  } else {
+    console.log("Session Token or Profile Data is Null. Setting profile data to null.");
+    localStorage.setItem('fh.wfm.profileData', null);
+    return deferred.reject();
+  }
+  return deferred.promise;
 };
 
+/**
+* Decrypt the users profile data using AES with a hash of the session key.
+*/
+
 var retrieveProfileData = function() {
-  var json = localStorage.getItem('fh.wfm.profileData');
-  return json ? JSON.parse(json) : null;
+  var deferred = q.defer();
+  if (localStorage.getItem('fh_session_token.sessionToken') === null) {
+    console.log("Error: Session Token is Undefined, Cannot Retrieve Profile Data.");
+    return deferred.reject(null);
+  } else if (localStorage.getItem('fh.wfm.profileData') === null) {
+    console.log("Error: Profile Data is Null. Cannot Retrieve Profile Data.");
+    return deferred.reject(null);
+  }
+
+  var ciphertext = localStorage.getItem('fh.wfm.profileData');
+  var sessionToken = JSON.parse(localStorage.getItem('fh_session_token.sessionToken')).sessionToken;
+
+  sha256.hash(sessionToken).then(function(hash) {
+    aes.decrypt(ciphertext, hash).then(function(plaintext) {
```

Comment:
Might be cleaner to do this? Again, this assumes a standard promise implementation.

```
sha256.hash(sessionToken)
  .then(function (hash) {
    return aes.decrypt(ciphertext, hash);
  });
```